src/diffTool/diffProcessor.ts

```diff
@@ -12,20 +12,42 @@ import { ramlToolLogger } from "../common/logger";
 import * as path from "path";
 import { applyRules } from "./rulesProcessor";
 
-export const diffRulesPath = path.join(__dirname, "../../resources/diff/rules");
+export const defaultRulesPath = path.join(
+  __dirname,
+  "../../resources/diff/rules/defaultRules.json"
+);
+
+/**
+ * Generate differences between two RAML files and apply rules
+ * @param leftRaml - Base RAML file to compare
+ * @param rightRaml - Other RAML file to compare with the left RAML to find differences
+ * @param rulesPath - Optional rules file to apply on the differences
+ *
+ * @returns Changes to the APIs
+ */
+export async function findApiChanges(
+  leftRaml: string,
+  rightRaml: string,
+  rulesPath?: string
```

Comment:
Why not `rulesPath = defaultRulesPath` or set it to the actual rule here?  Then you don't need the null check.